lib/data-types.js

```diff
@@ -460,6 +460,42 @@ class TIME extends ABSTRACT {
   toSql() {
     return 'TIME';
   }
+  validate(value) {
+    if (!Validator.isDate(String(value)) && !Validator.isDate(`01-01-1970 ${String(value)}`)) {
```

Comment:
```typescript
    if (value instanceof Date) {
      if (Number.isNaN(value.getTime())) {
        throw new Error()
      }
    }
    
    if (typeof value !== 'string) {
      throw new Error()
    }
    
    if (!/-?\d+:[0-5][0-9]:[0-5][0-9](?:\.\d+)?/.test(value)) {
      throw new Error()
    }

    return true;
```

While I'm fine with supporting `Date`, passing a date as a string should not be valid.

We should only either pass a `Date` object or a string that matches the format 
`-?\d+:[0-5][0-9]:[0-5][0-9](?:\.\d+)?`.

``Validator.isDate(`01-01-1970 ${String(value)}`)`` is also not the right way to check if a value is TIME as TIME can be greater than `23:59:59` and less than `00:00:00` when used as a duration.

Reply:
These changes have been made and the validator now accepts only dates object and time strings.